index.js

```diff
@@ -51,5 +51,155 @@ const state = {
+  storeItemImg.alt = state.items[i].name; //loop
+
+  //append the img element to div
+  storeItemDiv.appendChild(storeItemImg);
+
+  //create button
+  const addButton = document.createElement("button");
+  addButton.textContent = "Add to cart";
+
+  // Add event listener to "Add to cart" button
+  addButton.addEventListener("click", () => {
+    addToCart(i);
+  });
+
+  //append icon div and button to item
+  storeItem.appendChild(storeItemDiv);
+  storeItem.appendChild(addButton);
+
+  //find the element to put storeItem
+  var itemList = document.querySelector(".item-list.store--item-list");
+
+  //append the list item to the item list
+  itemList.appendChild(storeItem);
+}
+
+//function to render cart
+function renderCart() {
+  // Find the cart item list
+  const cartItemList = document.querySelector(".item-list.cart--item-list");
+  
+  // Clear the existing content of the cart
+  cartItemList.innerHTML = "";
+
+  //set total to equal 0 if no item is in the cart
+  const total = 0;
+  const totalElement = document.querySelector(".total-number");
+  totalElement.textContent = "£" + total.toFixed(2);
+
+  // Loop through the items in the cart
+  state.cart.forEach(item => {
+
+    // Create a new list item for the cart item
+    const cartItem = document.createElement("li");
+
+    // Create img element for the item icon
+    const cartItemImg = document.createElement("img");
+    cartItemImg.classList.add("cart--item-icon");
+    cartItemImg.src = "assets/icons/" + item.id + ".svg";
+    cartItemImg.alt = item.name;
+
+    // Create p element for the item name
+    const cartItemName = document.createElement("p");
+    cartItemName.textContent = item.name;
+
+    // Create button for decreasing quantity
+    const decreaseButton = document.createElement("button");
+    decreaseButton.classList.add("quantity-btn", "remove-btn", "center");
+    decreaseButton.textContent = "-";
+    decreaseButton.addEventListener("click", () => decreaseQuantity(item));
+
+    // Create span element for the item quantity
+    const cartItemQuantity = document.createElement("span");
+    cartItemQuantity.classList.add("quantity-text", "center");
+    cartItemQuantity.textContent = item.quantity;
+
+    // Create button for increasing quantity
+    const increaseButton = document.createElement("button");
+    increaseButton.classList.add("quantity-btn", "add-btn", "center");
+    increaseButton.textContent = "+";
+    increaseButton.addEventListener("click", () => increaseQuantity(item));
+
+    // Append all elements to the cart item
+    cartItem.appendChild(cartItemImg);
+    cartItem.appendChild(cartItemName);
+    cartItem.appendChild(decreaseButton);
+    cartItem.appendChild(cartItemQuantity);
+    cartItem.appendChild(increaseButton);
+
+    // Append the cart item to the cart item list
+    cartItemList.appendChild(cartItem);
+
+  // Update the total displayed on the webpage
+  const total = state.total.reduce((total, currentValue) => 
```

Comment:
The total calculation logic is placed inside the forEach loop, which means it tries to set the total text content multiple times unnecessarily. Move the total calculation and setting of the total text content outside of the forEach loop